### PR TITLE
add altnames #1165

### LINKED ISSUE #1165: [FEATURE REQUEST] Add alternative names
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### Problem

It would be much easier to find icons, if you could search for alternative names.

Take vscode for example.
Currently if I search for `Visual Studio Code`, or `visualstudiocode` nothing shows up. This is because the name of the icon is `vscode`.

The same is true for a lot of javascript frameworks and libraries, where they won't show up if you add `js` at the end

### Possible Solution

If an array of `altNames` was added to the devicon.json file, it could be added as a search field, and let us search through that list as well.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Snailedlt**
## Double check these details before you open a PR

- [x] PR does not match another non-stale PR currently opened

## Features

altNames added. Now users can search for alternative names of icons

**This PR closes NONE**

This PR closes #1165

### CODE REVIEW

**Comment by Snailedlt** [comment]
Contributing guidelines have to be updated to account for this before merging to master btw

File: devicon.json
Comment (amacado) [inline devicon.json:161]: why is androidstudio also named "arm64"? It's also a duplicate of "aarch64". This brings me to the point of: Should we allow duplicates for altnames? I would say "no", otherwise they would be used as tags, not alternative names. Or? So we would need a check for duplicate altnames.

File: devicon.json
Comment (amacado) [inline devicon.json:614]: Altname should not be allowed to be the same name as 'name' ;-) Need a check to adress this as well imo.

File: devicon.json
Comment (amacado) [inline devicon.json:738]: `bulma` is already added (see a couple lines below). This would add it a second time.

File: devicon.json
Comment (amacado) [inline devicon.json:1029]: The same applies for `clojurescript`, already added.

File: devicon.json
Comment (amacado) [inline devicon.json:1420]: `debian` is already the name, no need for an altname imo.

File: devicon.json
Comment (amacado) [inline devicon.json:1822]: `embeddedc` is already added. This would add it twice.

File: devicon.json
Comment (amacado) [inline devicon.json:2739]: Duplicate entry of `handlebars`

File: devicon.json
Comment (amacado) [inline devicon.json:3144]: This might be confusing/irretating or?

File: devicon.json
Comment (amacado) [inline devicon.json:3479]: Duplicate entry of `knockout`

File: devicon.json
Comment (amacado) [inline devicon.json:3525]: Empty string altnames should be forbidden?

File: devicon.json
Comment (amacado) [inline devicon.json:3784]: `md` is already used for `markdown`. Does it realy apply to `maya`?

File: devicon.json
Comment (amacado) [inline devicon.json:5552]: Duplicate entry for `rstudio`

File: devicon.json
Comment (amacado) [inline devicon.json:5957]: `subversion` and `sqllite`? I know subversion as a version management and sql lite as a database, how does this match?

File: devicon.json
Comment (amacado) [inline devicon.json:6177]: Duplicate entry for `storybook`

File: devicon.json
Comment (amacado) [inline devicon.json:6291]: Altname is the same as name.

File: devicon.json
Comment (amacado) [inline devicon.json:6397]: Duplicate entry `thealgorithms`

File: devicon.json
Comment (amacado) [inline devicon.json:6784]: `vuetify` has nothing in common with `visualstudiocode`

**Review by amacado — CHANGES_REQUESTED** [review]
There are a lot of errors and open questions to discuss before this pull request can be accepted. In addition to my in-line comments: The wiki/readme needs to be improved to explain the altname. 😄

File: devicon.json
Comment (Snailedlt) [reply devicon.json:161]: ohh, yes. Must have been a merge issue since I accidentally based the branch on master at first, then rebased on develop.

I will fix this soon :)

File: devicon.json
Comment (Snailedlt) [reply devicon.json:614]: Agreed. Not sure how to enforce it though. Maybe through github actions?

File: devicon.json
Comment (Snailedlt) [reply devicon.json:738]: Same as here: https://github.com/devicons/devicon/pull/1167#discussion_r873515186

File: devicon.json
Comment (Snailedlt) [reply devicon.json:1029]: https://github.com/devicons/devicon/pull/1167#discussion_r873515186

**Comment by Snailedlt** [comment]
@amacado Thanks for fishing out a lot of errors.
I will take a look at fixing it soon. Might have to redo most of it by hand, since there are so many errors due to rebasing :/

When it comes to updating the wiki, I agree. But should we wait with that until it's released, or do it now?

**Comment by amacado** [comment]
>When it comes to updating the wiki, I agree. But should we wait with that until it's released, or do it now?

Let's wait till it's released.. I've found this pages which needs an update:
- https://github.com/devicons/devicon/wiki/Example-of-Submitting-Icons-to-the-Repository
- https://github.com/devicons/devicon/wiki/Updating-the-%60devicon.json%60

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: fixed: 3c5bf839850823623296b0f0a222ba65aff6386c

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 3c5bf839850823623296b0f0a222ba65aff6386c

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 3c5bf839850823623296b0f0a222ba65aff6386c

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 3c5bf839850823623296b0f0a222ba65aff6386c

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 3c5bf839850823623296b0f0a222ba65aff6386c

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 3c5bf839850823623296b0f0a222ba65aff6386c

File: devicon.json
Comment (Snailedlt) [reply devicon.json:3110]: Jenkins is a fork of hudson. So hudson is the old name, but they had to change it due to legal issues with Oracle.

You can read more about it here: https://www.javatpoint.com/jenkins-vs-hudson#:~:text=Jenkins%20vs%20Hudson%20Comparison%20Table,be%20forked%20and%20Jenkins%20CI.

File: devicon.json
Comment (Snailedlt) [reply devicon.json:3491]: Agreed

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 3c5bf839850823623296b0f0a222ba65aff6386c

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 6e9a895871f9c489551e347f4b9c05ee0830f8ae

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 6e9a895871f9c489551e347f4b9c05ee0830f8ae

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 6e9a895871f9c489551e347f4b9c05ee0830f8ae

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: Nope, another mistake.
Fixed here: 6e9a895871f9c489551e347f4b9c05ee0830f8ae

File: devicon.json
Comment (Snailedlt) [reply devicon.json]: 6e9a895871f9c489551e347f4b9c05ee0830f8ae

File: devicon.json
Comment (Snailedlt) [reply devicon.json:3078]: So imo it should stay, but guess it's up for discussion

**Comment by Snailedlt** [comment]
@amacado I fixed all the issues you found, and made some more finishing touches. Take another look when you have time :)

File: docs/assets/js/script.js
Comment (maltejur) [inline docs/assets/js/script.js:14]: This should probably not be changed

File: docs/assets/js/script.js
Comment (maltejur) [inline docs/assets/js/script.js:27]: this too

**Review by maltejur — CHANGES_REQUESTED** [review]
The actual `devicon.json` changes look fine to me, it just seems some temporary code changes from you slipped through.

File: docs/assets/js/script.js
Comment (Snailedlt) [reply docs/assets/js/script.js:14]: nice catch. Yes definitely should be changed

Comment (Snailedlt) [reply docs/assets/js/script.js]: fixed: 726d58f1bcadbd4da6910e041d0bf5a5d3ec092b

File: docs/assets/js/script.js
Comment (Snailedlt) [reply docs/assets/js/script.js]: fixed: 726d58f1bcadbd4da6910e041d0bf5a5d3ec092b

**Review by maltejur — APPROVED** [review]

**Comment by Snailedlt** [comment]
How many maintainers usually need to approve before PRs can be merged to develop?

**Review by amacado — APPROVED** [review]

**Pull request merged**
